storage/innobase/handler/ha_innodb.cc

```diff
@@ -14822,12 +14821,34 @@ ha_innobase::check(
 	ulint		n_rows_in_table	= ULINT_UNDEFINED;
 	bool		is_ok		= true;
 	dberr_t		ret;
+        uint handler_flags= check_opt->handler_flags;
```

Comment:
This is something that @montywi developed. I am not very familiar with the `CHECK TABLE` code outside InnoDB, and until working on this bug I did not even know that it can rewrite the `.frm` file. It seems to me that implemneting a cleaner solution would require a major refactoring of this API, which could require a nontrivial amount of work in every storage engine, including any possibly externally maintained ones. That would not be something that we would want to do in maintenance releases.

Reply:
ok, I just wanted to mention it but don't want to hold the review for it. Let me know if you get more information on it.